exercises/variables/variables5.rs

```diff
@@ -1,11 +1,11 @@
 // variables5.rs
 // Execute `rustlings hint variables5` or use the `hint` watch subcommand for a hint.
 
-// I AM NOT DONE
-
 fn main() {
     let number = "T-H-R-E-E"; // don't change this line
     println!("Spell a Number : {}", number);
-    number = 3; // don't rename this variable
-    println!("Number plus two is : {}", number + 2);
+    {
+        let number = 3; // don't rename this variable
+        println!("Number plus two is : {}", number + 2);
+    }
```

Comment:
Admittedly while the hint points to using shadowing but this actually isn't required. You can just "re-"let and go on.

I don't know if that was the goal of the instruction though as the comments don't say. "don't change this line" just not to rename it.

Although shadowing is an interesting concept. Creating a closure in the middle of a scope is both nice and dangerous.

Reply:
Huh...I guess I'd be curious what the intention was you know what I mean? I get the impression the "spiritual goal" of shadowing is kinda like closures in JS - mutation in a "safe space" that doesn't pollute the parent scope. Good call out though.

Reply:
I am curious as to why there is such a focus on variable name reuse personally. I am interested in the real world reason I might wanna ascribe a new type or value to the same variable regardless of the side-effect duration.

Reply:
Well, if I had to guess, it's related to the whole "naming things is hard" thing. If you could re-use variables in a way that is safe and cogent, you can let `number` be **the** thing instead of **a** thing. Like I said, I'm not totally sold on it, but I can see at first blush the value.